ui/lib/core/addon/components/confirm-action.js

```diff
@@ -14,50 +14,77 @@ import hbs from 'htmlbars-inline-precompile';
  *  </ConfirmAction>
  * ```
  *
- * @property {Func} onConfirmAction=null - The action to take upon confirming.
- * @property {String} [confirmMessage=Are you sure you want to do this?] - The message to display upon confirming.
+ * @property {Func} [onConfirmAction=null] - The action to take upon confirming.
+ * @property {String} [confirmTitle=Delete this?] - The title to display when confirming.
+ * @property {String} [confirmMessage=Are you sure you want to do this?] - The message to display when confirming.
  * @property {String} [confirmButtonText=Delete] - The confirm button text.
  * @property {String} [cancelButtonText=Cancel] - The cancel button text.
+ * @property {String} [disabledTitle=Can't delete this yet] - The title to display when the button is disabled.
  * @property {String} [disabledMessage=Complete the form to complete this action] - The message to display when the button is disabled.
  *
  */
 
 export default Component.extend({
-  tagName: 'span',
   classNames: ['confirm-action'],
   layout: hbs`
-    {{#if showConfirm ~}}
-      <span class={{containerClasses}}>
-        <span class={{concat 'confirm-action-text ' messageClasses}}>{{if disabled disabledMessage confirmMessage}}</span>
-        <button {{action 'onConfirm'}} disabled={{disabled}} class={{confirmButtonClasses}} type="button" data-test-confirm-button=true>{{confirmButtonText}}</button>
-        <button {{action 'toggleConfirm'}} type="button" class={{cancelButtonClasses}} data-test-confirm-cancel-button=true>{{cancelButtonText}}</button>
-      </span>
-    {{else}}
-      <button
-        class={{buttonClasses}}
+    {{#basic-dropdown class="popup-menu" horizontalPosition=horizontalPosition verticalPosition=verticalPosition onOpen=(action "toggleConfirm") onClose=(action "toggleConfirm") as |d|}}
+      {{#d.trigger
+        tagName="button"
         type="button"
-        disabled={{disabled}}
-        data-test-confirm-action-trigger=true
-        {{action 'toggleConfirm'}}
-      >
+        class=(concat buttonClasses " popup-menu-trigger" (if d.isOpen " is-active"))
+        disabled=disabled
+        data-test-confirm-action-trigger="true"
+      }}
         {{yield}}
-      </button>
-    {{~/if}}
+        {{#if (eq buttonClasses 'toolbar-link')}}
+          <Icon @glyph="chevron-{{if showConfirm 'up' 'down'}}" />
+        {{/if}}
+      {{/d.trigger}}
+      {{#d.content class=(concat "popup-menu-content")}}
+        <div class="box confirm-action-message">
```

Comment:
Could we use one of the Alert components here? It looks like AlertBanner will take a title and a message to display, but not sure if the styling matches.

Reply:
I thought about it, but yeah the styling doesn't match. Any small change in the styling (like padding) of Alerts would be felt much more here, so I figured this was better.